FirebaseAdmin/FirebaseAdmin/Auth/UserImportRequest.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace FirebaseAdmin.Auth
+{
+    /// <summary>
+    /// Encapsulates user import requests by specifying hashing properties for passwords and
+    /// the list of users to be imported.
+    /// </summary>
+    internal class UserImportRequest
+    {
+        internal const int MaxImportUsers = 1000;
+
+        [JsonProperty("users")]
+        private readonly IEnumerable<ImportUserRecordArgs.ImportUserRequest> users;
```

Comment:
I suppose this can also be a property with a private setter?

Reply:
That makes sense, although I think the readonly modifier is lost this way

```
internal IEnumerable<ImportUserRecordArgs.ImportUserRequest> Users { get; private set; }
```